src/main/utils/shell-env.ts

```diff
@@ -3,11 +3,85 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 
-import shellEnvironment from "shell-env";
+import { spawn } from "child_process";
+import { randomUUID } from "crypto";
+import { basename } from "path";
+import { isWindows } from "../../common/vars";
 import logger from "../logger";
 
 export type EnvironmentVariables = Partial<Record<string, string>>;
 
+
+async function unixShellEnvironment(shell: string): Promise<EnvironmentVariables> {
+  const runAsNode = process.env["ELECTRON_RUN_AS_NODE"];
+  const noAttach = process.env["ELECTRON_NO_ATTACH_CONSOLE"];
+  const env = {
+    ...process.env,
+    ELECTRON_RUN_AS_NODE: "1",
+    ELECTRON_NO_ATTACH_CONSOLE: "1",
+  };
+  const mark = randomUUID().replace(/-/g, "");
+  const regex = new RegExp(`${mark}(.*)${mark}`);
+  const shellName = basename(shell);
+  let command: string;
+  let shellArgs: string[];
+  
+  if (/^pwsh(-preview)?$/.test(shellName)) {
+    // Older versions of PowerShell removes double quotes sometimes so we use "double single quotes" which is how
+    // you escape single quotes inside of a single quoted string.
+    command = `& '${process.execPath}' -p '''${mark}'' + JSON.stringify(process.env) + ''${mark}'''`;
+    shellArgs = ["-Login", "-Command"];
+  } else {
+    command = `'${process.execPath}' -p '"${mark}" + JSON.stringify(process.env) + "${mark}"'`;
```

Comment:
Basically we outsource environment variable parsing to a nodejs process (which is started using shell environment). Markers are needed because shell might output some intro text that would confuse `JSON.parse`.